catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/associations/Associated.java

```diff
@@ -134,9 +135,10 @@ public void putAssociations(String id, Collection<Edge> edges)
             return;
         }
 
-        catalogFramework.update(new UpdateRequestImpl(changedMetacards.keySet()
-                .toArray(new String[0]), new ArrayList<>(changedMetacards.values())));
-
+        catalogFramework.update(new UpdateRequestImpl(new HashSet<>(writableSourceIds),
+                changedMetacards.keySet()
+                        .toArray(new String[0]),
```

Comment:
I believe it's more efficient to create the string array with expected size instead of zero.